src/transformers/models/wav2vec2/processing_wav2vec2.py

```diff
@@ -102,7 +105,22 @@ def from_pretrained(cls, pretrained_model_name_or_path, **kwargs):
                 :class:`~transformers.PreTrainedTokenizer`
         """
         feature_extractor = Wav2Vec2FeatureExtractor.from_pretrained(pretrained_model_name_or_path, **kwargs)
-        tokenizer = Wav2Vec2CTCTokenizer.from_pretrained(pretrained_model_name_or_path, **kwargs)
+
+        # load generic `AutoTokenizer`
+        # need fallback here for backward compatibility in case processor is
+        # loaded from just a tokenizer file that does not have a `tokenizer_class` attribute
+        # behavior should be deprecated in major future release
+        try:
+            tokenizer = AutoTokenizer.from_pretrained(pretrained_model_name_or_path, **kwargs)
+        except OSError:
+            warnings.warn(
+                f"Loading a tokenizer inside {cls.__name__} from a config that does not"
+                " include a `tokenizer_class` attribute is deprecated and will be removed in v5. Please add the following "
+                " attribute to your `tokenizer_config.json` file to suppress this warning: "
+                " `'tokenizer_class': 'Wav2Vec2CTCTokenizer'`",
```

Comment:
I don't remember we deprecated this behavior. There is nothing in the `AutoTokenizer` class for that.

Reply:
Hmm - maybe the warning isn't super clear. But the reason is the following:

We need to use the `AutoTokenizer` instead of `Wav2Vec2CTCTokenizer` as `Wav2Vec2CTCTokenizer` is too restrictive. Now there is one use case (which is more or less never used IMO) where this change is backwards breaking:

If one wants to load a `Wav2Vec2Processor` from a directory that only includes:

- `preprocessing.json`
- `tokenizer_config.json`

file but no model `config.json` & the `tokenizer_config.json` file doesn't define which class to load. 
In this case `AutoTokenizer` cannot find the correct tokenizer class. `Wav2Vec2CTCTokenizer` would have found the correct class however which is why this `try .... except ...` statement is added here with a deprecation warning though.

Reply:
The warning should be adapted to say there should be either a config.json or a tokenizer_config.json with that attribute then :-)